qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -784,10 +824,17 @@ def _run_analysis(
         # pop arguments that are not given to fitter
         extra_options = self._arg_parse(**options)
 
-        # TODO update this with experiment metadata PR #67
         try:
-            self.__qubits = experiment_data.data(0)["metadata"]["physical_qubits"]
-        except KeyError:
+            metadata = experiment_data.metadata()
+            self.__experiment_type = metadata["experiment_type"]
+            self.__qubits = metadata["physical_qubits"]
+
+            latest_job_metadata = metadata["job_metadata"][-1]
+            self.__experiment_options = latest_job_metadata["experiment_options"]
+            self.__analysis_options = latest_job_metadata["analysis_options"]
+            self.__run_options = latest_job_metadata["run_options"]
+            self.__transpile_options = latest_job_metadata["transpile_options"]
```

Comment:
Are we always guaranteed to have these variables? Is this covered by some of our tests.

Reply:
Yes. These are values defined by the experiment base class. These values are guaranteed unless you write experiment class from scratch.

Reply:
FYI
https://github.com/Qiskit/qiskit-experiments/blob/b2c9ff7d4780b0c629ff4f643be6ee1e9b8b015c/qiskit_experiments/base_experiment.py#L342-L357